src/EventListener/AssignLocaleListener.php

```diff
@@ -44,14 +33,20 @@ public function prePersist(EventArgs $args): void
     private function assignLocale(EventArgs $args): void
     {
         $object = $args->getObject();
+        $localeCode = $this->translator->loadCurrentLocale();
+
+        if ($object instanceof TranslatableInterface) {
+            $object->setCurrentLocale($localeCode);
+            $object->setFallbackLocale($this->defaultLocale);
 
-        if (!$object instanceof TranslatableInterface) {
             return;
         }
 
-        $localeCode = $this->translator->loadCurrentLocale();
+        // fill the locale where it's missing
+        if ($object instanceof TranslationInterface && !$object->getLocale()) {
+            $object->setLocale($localeCode);
```

Comment:
@SpartakusMd why did you set locale here?

Reply:
It was done in order to fill it automatically when it's missing. Don't remember specifically the case when I found it when used the package inside the project. Maybe not needed.

Reply:
@SpartakusMd if the config and translation classes are all setup then locale should always exist. So I don't think this line is needed. If you manage to find the case where it happened you can open a new issue and we can check it out :)

Reply:
Sure. Made the change.